daemon/src/projection.rs

```diff
@@ -790,8 +791,9 @@ impl Tx {
                     &a.aggregated.creation_timestamp,
                 )
             })
-            .collect();
+            .collect::<Vec<_>>();
 
+        update_position_metrics(cfds_with_quote.as_slice());
```

Comment:
`projection` was designed to be the read model that is exposed to the UI. Retro-fitting metrics into here is not ideal.

It shouldn't be too difficult to create a separate actor that periodically loads all CFDs and exposes these metrics. That way, you can extract exactly the information you want and don't have to re-purpose the `projection::Cfd` model.

Reasoning from an existing read-model is likely to produce all sorts of bugs.

I am worried that adding these here will make `projection` significantly harder to maintain as it its model serves yet another purpose now.

Reply:
The problem with a separate actor is that we can't update the metrics in real-time. However, is is more a limitation of the fact that our ES implementation doesn't have event subscriptions (i.e. notifying actors about a new event).

At the moment, we circumvent that by notifying `projection` explicitly at end of `process_manager`. We could easily to the same thing for a `metrics` actor and thus avoid loading things periodically.